consistency_check/src/bin/jsons/main.rs

```diff
@@ -176,46 +167,45 @@ async fn check_jsons_consistency(rocks_path: String, postgre_creds: String, batc
     let rate = Arc::new(Mutex::new(0.0));
     let count_of_missed_jsons = Arc::new(AtomicU64::new(0));
 
-    let shutdown_token = CancellationToken::new();
-    let (shutdown_for_file_writer_tx, shutdown_for_file_writer_rx) = broadcast::channel::<()>(1);
-
-    let mut writers = JoinSet::new();
-
     let missed_jsons: Arc<Mutex<HashSet<String>>> = Arc::new(Mutex::new(HashSet::new()));
 
     info!("Launching writer job...");
     let missed_jsons_cloned = missed_jsons.clone();
-    writers.spawn(async move {
-        let missed_jsons_file =
-            File::create(MISSED_JSONS_FILE).expect("Failed to create file for missed jsons");
+    usecase::executor::spawn({
+        let cancellation_token = cancellation_token.child_token();
+        async move {
+            let missed_jsons_file =
+                File::create(MISSED_JSONS_FILE).expect("Failed to create file for missed jsons");
+
+            let mut missed_jsons_wrt = csv::Writer::from_writer(missed_jsons_file);
+
+            loop {
+                let mut f_ch = missed_jsons_cloned.lock().await;
+                for t in f_ch.iter() {
+                    missed_jsons_wrt.write_record([t]).unwrap();
+                }
+                missed_jsons_wrt.flush().unwrap();
+                f_ch.clear();
 
-        let mut missed_jsons_wrt = csv::Writer::from_writer(missed_jsons_file);
+                drop(f_ch);
 
-        loop {
-            let mut f_ch = missed_jsons_cloned.lock().await;
-            for t in f_ch.iter() {
-                missed_jsons_wrt.write_record([t]).unwrap();
-            }
-            missed_jsons_wrt.flush().unwrap();
-            f_ch.clear();
-
-            drop(f_ch);
+                if cancellation_token.is_cancelled() {
```

Comment:
The same question as above. Does it mean all `write_record` happen anyway even if shutdown signal is received?

Reply:
yes, as it did before